routes/trackRouter.js

```diff
@@ -1074,34 +1074,34 @@ router.get('/', function (req, res, next) {
                 if (req.app.locals.main.result) {
                     return res.render("track", req.app.locals.main.result);
                 } else {
-                    // TODO: at this point, user's credentials are valid, but they do not have custom curated tracks
-                    //  display an 'error' page, which tells user they need to take a photo, and has a button prompting to go back to homepage
-                    return res.render('error', {
-                        message: "No custom tracks yet, please take a photo to get curated tracks",
-                        error: {status: 400, stack: "no stack"}
-                    });
+                    return res.redirect("/webcam/#/error/no-photo-submitted");
                 }
             } else {
                 return res.redirect("/spotify/login");
             }
         })
         .catch((err) => {
-            // TODO: just in case checkCredentials throws an error (highly unlikely)
-            console.log(err);
+            return res.redirect("/spotify/login");
         })
 });
 
 // REQUIRES: req.body to contain a list of track URI's in format ["spotify:track:1ue7zm5TVVvmoQV8lK6K2H", ...]
 router.post('/', function (req, res, next) {
     let main = req.app.locals.main;
-    return main.createMoodPlaylist(req.body)
-        .then((playlistURL) => {
-            return res.status(200).json({link: playlistURL});
-        })
-        .catch((err) => {
-            console.log(err);
-            return res.status(400).json({"error": err.message});
-        })
+
+    if(main) {
+        return main.createMoodPlaylist(req.body)
```

Comment:
Also should we check that req.body is properly formatted?

Reply:
Hmm, I think that's more so the responsibility of the backend? If it fails the check inside Main then the error will propagate back up- what are your thoughts?

Reply:
Hmm, I think that's more so the responsibility of the backend? If it fails the check inside Main then the error will propagate back up- what are your thoughts?

Reply:
OHya good point yeah